Extension/src/LanguageServer/ui_new.ts

```diff
@@ -0,0 +1,548 @@
+        } else {
+            items.push({ label: localize({ key: "pause.analysis", comment: [this.codeAnalysisTranslationHint]}, "Pause {0}", this.codeAnalysisProgram), description: "", index: 1 });
+        }
+        items.push({ label: localize({ key: "another.analysis", comment: [this.codeAnalysisTranslationHint]}, "Start Another {0}...", this.codeAnalysisProgram), description: "", index: 3 });
+        const selection: IndexableQuickPickItem | undefined = await vscode.window.showQuickPick(items, options);
+        return (selection) ? selection.index : -1;
+    }
+
+    public async showIdleCodeAnalysisCommands(): Promise<number> {
+        const options: vscode.QuickPickOptions = {};
+        options.placeHolder = this.selectACommandString;
+
+        const items: IndexableQuickPickItem[] = [];
+        items.push({ label: localize({ key: "active.analysis", comment: [this.codeAnalysisTranslationHint]}, "Run Code Analysis on Active File", this.codeAnalysisProgram), description: "", index: 0 });
+        items.push({ label: localize({ key: "all.analysis", comment: [this.codeAnalysisTranslationHint]}, "Run Code Analysis on All Files", this.codeAnalysisProgram), description: "", index: 1 });
+        items.push({ label: localize({ key: "open.analysis", comment: [this.codeAnalysisTranslationHint]}, "Run Code Analysis on Open Files", this.codeAnalysisProgram), description: "", index: 2 });
+        const selection: IndexableQuickPickItem | undefined = await vscode.window.showQuickPick(items, options);
+        return (selection) ? selection.index : -1;
+    }
+
+    public showConfigureIncludePathMessage(prompt: () => Promise<boolean>, onSkip: () => void): void {
+        setTimeout(() => {
+            this.showConfigurationPrompt(ConfigurationPriority.IncludePath, prompt, onSkip);
+        }, 10000);
+    }
+
+    public showConfigureCompileCommandsMessage(prompt: () => Promise<boolean>, onSkip: () => void): void {
+        setTimeout(() => {
+            this.showConfigurationPrompt(ConfigurationPriority.CompileCommands, prompt, onSkip);
+        }, 5000);
+    }
+
+    public showConfigureCustomProviderMessage(prompt: () => Promise<boolean>, onSkip: () => void): void {
+        this.showConfigurationPrompt(ConfigurationPriority.CustomProvider, prompt, onSkip);
+    }
+
+    private showConfigurationPrompt(priority: ConfigurationPriority, prompt: () => Thenable<boolean>, onSkip: () => void): void {
+        const showPrompt: () => Promise<ConfigurationStatus> = async () => {
+            const configured: boolean = await prompt();
+            return Promise.resolve({
+                priority: priority,
+                configured: configured
+            });
+        };
+
+        if (this.curConfigurationStatus) {
+            this.curConfigurationStatus = this.curConfigurationStatus.then(result => {
```

Comment:
We're trying to avoid use of `then` if not absolutely necessary, in favor of `async`/`await`.  Could you make this an async function and await here?  If you don't await `showConfigurationPrompt`, the effect should be the same.

Reply:
This is current code, as discussed we'll address this separately.